cloudformation/template.yaml

```diff
@@ -103,6 +103,8 @@ Parameters:
     Type: String
     Description: If you provided a domain name associated with an acm cert, then you must also specify here the acm cert's arn. Leave this blank to create a developer portal without a custom domain name.
     Default: ''
+    AllowedPattern: ^$|^arn:aws:acm:(ap-east-1|ap-northeast-1|ap-northeast-2|ap-northeast-3|ap-south-1|ap-southeast-1|ap-southeast-2|ca-central-1|cn-north-1|cn-northwest-1|eu-central-1|eu-north-1|eu-west-1|eu-west-2|eu-west-3|sa-east-1|us-east-1|us-east-2|us-gov-east-1|us-gov-west-1|us-west-1|us-west-2){1}:([0-9]){12}:certificate\/[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}$
```

Comment:
I'm really leery of explicit lists of regions; I think we should trust the underlying service (ACM) with the validation here. Maybe instead we could use:
`^$|^arn:aws:acm:[a-z0-9-]+:([0-9]){12}:certificate\/[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}$`

Downside is this will let some invalid acm certs start a deployment that's guaranteed to fail, but the upside is it future-proofs the regex & won't need updated.

Reply:
I completely agree.
If you're happy with letting some duff regions through then I could tighten it up to at least follow the current naming pattern:

`^$|^arn:aws:acm:([a-z]){2}((-[a-z]+)){1,2}-([0-9]){1}:([0-9]){12}:certificate\/[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}$`

Reply:
Might as well use the arn pattern from the [ACM docs](https://docs.aws.amazon.com/acm/latest/APIReference/API_GetCertificate.html#API_GetCertificate_RequestSyntax): `arn:[\w+=/,.@-]+:[\w+=/,.@-]+:[\w+=/,.@-]*:[0-9]+:[\w+=,.@-]+(/[\w+=,.@-]+)*`

Reply:
> Might as well use the arn pattern from the [ACM docs](https://docs.aws.amazon.com/acm/latest/APIReference/API_GetCertificate.html#API_GetCertificate_RequestSyntax): `arn:[\w+=/,.@-]+:[\w+=/,.@-]+:[\w+=/,.@-]*:[0-9]+:[\w+=,.@-]+(/[\w+=,.@-]+)*`

This is the regex for any ARN though. The validation could be replaced with this and it would definitely catch simple mistakes in the formatting. But would it not be better to validate that the ARN is in the correct format for a certificate?
Possibly the region part of the regex from the generic ARN regex could be used:

`^$|^arn:aws:acm:[\w+=/,.@-]*:([0-9]){12}:certificate\/[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}$`

Reply:
> > Might as well use the arn pattern from the [ACM docs](https://docs.aws.amazon.com/acm/latest/APIReference/API_GetCertificate.html#API_GetCertificate_RequestSyntax): `arn:[\w+=/,.@-]+:[\w+=/,.@-]+:[\w+=/,.@-]*:[0-9]+:[\w+=,.@-]+(/[\w+=,.@-]+)*`
> 
> This is the regex for any ARN though. The validation could be replaced with this and it would definitely catch simple mistakes in the formatting. But would it not be better to validate that the ARN is in the correct format for a certificate?
> Possibly the region part of the regex from the generic ARN regex could be used:
> 
> `^$|^arn:aws:acm:[\w+=/,.@-]*:([0-9]){12}:certificate\/[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}$`

I've updated the code to use this last piece of regex.
It uses the region validation from the generic arn validation regex but enforces the rest of the arn to match that of a certificate.